pkg/cloudprovider/cloudprovider.go

```diff
@@ -317,6 +317,9 @@ func (c *CloudProvider) instanceToNodeClaim(i *instance.Instance, instanceType *
 	}
 	labels[v1.LabelTopologyZone] = i.Zone
 	labels[corev1beta1.CapacityTypeLabelKey] = i.CapacityType
+	if v, ok := i.Tags[v1beta1.LabelNodeClass]; ok {
```

Comment:
Does this take into account the scheduling behavior of this label being there? What if someone has an affinity for a certain NodeClass label? Are we going to take this into account?

Reply:
Added in latest commit